### ISSUE TITLE
Examples should not be LGPLd

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7205]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)| @jyknight reported|
|-|-|
|Trac ID|trac#361|
|Type|enhancement|
|Created|2003-10-20 23:12:03Z|

Attachments:

* [license.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b6e/b6e7a22c25e9e7146913841b0995288c688f9bf8/930ff9e732755d92ad4ffb21bc49b7e5e3e6ad46.txt) (2467 bytes) - added by jknight on 2003-10-20 23:12:03Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__361 361
type__enhancement enhancement
reporter__jknight jknight
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1066691523000000 1066691523000000
changetime__1469002708497245 1469002708497245
version__ 
owner__ 
cc__glyph cc__spiv cc__jknight cc__hypatia
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
The examples seem to be under the LGPL, like the rest of the project. Some (31) explicitly state 
that they are, others (36) have no license text at all. That means anyone who 
wants to base their code on the examples is contaminated by the LGPL. This is a bad situation, 
unless the contamination is intentional, but that doesn't seem likely for a package whose overall 
license is LGPL. "Your code can use any license as long as you didn't start with examples we 
provided" doesn't really make much sense to me.

Everything in the examples directory, and the examples in the documentation should be explicitly 
relicensed under some unrestrictive license. 

As one possibility (not explicitly recommended -- it's just the first license of the sort I 
remembered), I attached Apple's license for their example code. It seems to essentially say that 
you may do anything with the code, but if you redistribute it in whole, you must include the notice.
Probably a simpler license could be used by Twisted.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
It's been like this for ages, it can stand to wait a bit more.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
This is, at best, a feature request.
Downgrading.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Eh, if we're going to change the "license" of the examples at all, i'd recommend
PD (or PD-equivalent, for those who don't think explicitly stating PD is legal)

But, I also think it's not really worth worrying about until someone who it
actually affects complains.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
I'm not really sure about the issues regarding putting something into public domain...from what I 
can tell, it looks as simple as putting "This file is granted to the public domain." instead of 
"Copyright 2003, xyz".

As for the "not worry about it until someone complains", that's basically saying that you don't 
mean the license you state. If that's the case why not just change, and be clear about what 
permission you really do intend to give people. I find it likely that people will ignore the fact 
that examples are LGPLd and use them anyways than complain about it.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
see the section on the public domain: http://zooko.com/license_quick_ref.html

I think it's reasonable to keep the examples LGPL because I can't see anyone
sharing enough code with the distributed examples to be considered a "derivative
work".  If you start from one of the examples, but then change everything,
you're not beholden to any particular copyright law, license or no license.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
So PD is out, too confusing.

As to the second, I'm pretty sure that isn't true. If you start with a piece of code and modify it, your 
work is considered a derivative of it, even if you later modify your work such that the original is no 
longer recognizable. Of course, as I'm no lawyer, I could be wrong, but why leave it up in the air like 
that?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
examples are now MIT! hooray. resolving.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|GitHub commented|
|-|-|

In changeset 958ecfe2894daf9f58f530f8adf8de0b2a69f33c
```
#!CommitTicketReference repository="" revision="958ecfe2894daf9f58f530f8adf8de0b2a69f33c"
Merge pull request #361 from twisted/8627-markrwilliams-universalpicklingerror-docstring

Author: markrwilliams
Reviewer: rodrigc
Fixes: #8627

Fix docstring in Pickle tests
```
